processor/stream/processor.go

```diff
@@ -307,3 +309,16 @@ func (sr *streamReader) wrapError(err error) error {
 	}
 	return err
 }
+
+// copyEvent returns a shallow copy of the APMEvent with a deep copy of the
+// labels and numeric labels.
+func copyEvent(e model.APMEvent) model.APMEvent {
```

Comment:
so, it makes sense that we need to make a deep copy and not re-use the labels between decodings.

What I still can't answer is "how did this work before?" The bug surfaced when we moved from using `common.MapStr` to `map[string](NumericLabelValue|LabelValue)`([pr](https://github.com/elastic/apm-server/pull/6682)), so I don't know if the map is 100% at fault. Reverting the above PR resulted in the bug no longer surfacing.